cmd/client/main.go

```diff
@@ -235,6 +243,10 @@ func main() {
 	logger := promlog.New(&promlogConfig)
 	coordinator := Coordinator{logger: logger}
 
+	if *myFqdn == "" {
+		level.Error(coordinator.logger).Log("msg", "automatic FQDN discovery failed.  use --fqdn.")
```

Comment:
This can also happen if the user does `--fqdn=""`, so we can make the error a little more generic.

```suggestion
		level.Error(coordinator.logger).Log("msg", "Missing FQDN hostname, use --fqdn.")
```